src/openssl.c

```diff
@@ -4693,9 +4754,20 @@ static int pk_toPEM(lua_State *L) {
 
 			break;
 		case 2: case 3: /* private, PrivateKey */
-			if (!PEM_write_bio_PrivateKey(bio, key, 0, 0, 0, 0, 0))
+			if (cname) {
+				cipher = EVP_get_cipherbyname(cname);
+				if (!cipher)
+					return luaL_error(L, "pkey:toPEM: unknown cipher: %s", cname);
+				if (!getfield(L, i, "password"))
+					return luaL_error(L, "pkey:toPEM: password not defined");
+			}
+
+			if (!PEM_write_bio_PrivateKey(bio, key, cipher, NULL, 0, pem_pw_cb, L))
```

Comment:
if there is no cname on the stack then what will be at top of stack?

Reply:
Added an else branch which pushes a nil value. The value will not be used because openssl will not invoke the callback when cipher equals NULL.